cg/meta/workflow/mutant/metadata_parser/utils.py

```diff
@@ -0,0 +1,53 @@
+from cg.constants.constants import ControlOptions
+from cg.meta.workflow.mutant.metadata_parser.models import SampleMetadata
+from cg.store.models import Sample
+from cg.apps.lims.api import LimsAPI
+from cg.constants.lims import LimsArtifactTypes, LimsProcess
+from genologics.entities import Artifact
+from genologics.entities import Sample as LimsSample
+from cg.models.cg_config import LOG
+
+
+def is_sample_external_negative_control(sample: Sample) -> bool:
+    return sample.control == ControlOptions.NEGATIVE
+
+
+def get_negative_controls_from_list(samples: list[LimsSample]) -> list[LimsSample]:
+    """Filter and return a list of internal negative controls from a given sample list."""
+    negative_controls = []
+    for sample in samples:
+        if sample.udf.get("Control") == "negative" and sample.udf.get("customer") == "cust000":
+            negative_controls.append(sample)
+    return negative_controls
+
+
+def get_internal_negative_control_id_from_lims(
+    lims: LimsAPI, sample_internal_id: str
+) -> None | str:
+    artifact: Artifact = lims.get_latest_artifact_for_sample(
+        LimsProcess.COVID_POOLING_STEP, LimsArtifactTypes.ANALYTE, sample_internal_id
+    )
+    if not artifact:
+        return None
+    samples = artifact[0].samples
+
+    negative_controls: list = get_negative_controls_from_list(samples=samples)
+
+    if len(negative_controls) > 1:
+        sample_ids = [sample.id for sample in negative_controls]
+        LOG.warning(f"Several internal negative control samples found: {' '.join(sample_ids)}")
+    else:
+        return negative_controls[0].id
+
+
+def get_internal_negative_control_id(lims: LimsAPI, metadata_for_case: SampleMetadata) -> str:
```

Comment:
I agree it gets confusing especially when it comes to lims data structures. Does the docstring I added help? If not, do you have any further suggestions of how I could make it more understandable?

Reply:
Let's first move all these methods to the LIMS API, and move the general ones like `has_valid_total_reads` to a QC service. I think we already have something defined

Reply:
We have the `SequencingQCService`. It would be nice to either implement something similar for Mutant or reuse the existing one if they share common methods. There are also several utility methods in that folder 🤔

Reply:
I moved the functions to query lims to the limsAPI and now use the `SequencingQCService` for all the QC checks!

Reply:
amazing!!